frame/support/procedural/src/pallet/parse/extra_constants.rs

```diff
@@ -50,14 +51,40 @@ pub struct ExtraConstantDef {
 	pub type_: syn::Type,
 	/// The doc associated
 	pub doc: Vec<syn::Lit>,
+	/// Optional MetaData Name
+	pub metadata_name: Option<syn::Ident>,
```

Comment:
actually I would rather override the ident field with the new name, because ident is only use for the name in the metadata. But anyway is fine to me.

like something like this:
```diff
--- a/frame/support/procedural/src/pallet/parse/extra_constants.rs
+++ b/frame/support/procedural/src/pallet/parse/extra_constants.rs
@@ -45,14 +45,12 @@ pub struct ExtraConstantsDef {
 
 /// Input definition for an constant in pallet.
 pub struct ExtraConstantDef {
-       /// Name of the function
-       pub ident: syn::Ident,
+       /// Name of the constant (function name of set by attribute)
+       pub name: syn::Ident,
```

but I'm ok as it is.

Reply:
Are you suggesting removal of the new `metadata_name` field entirely and just use an altered `name: Ident` field?

The problem is that `extra_constants` need both of those data because the expanded constants need to know which function it is actually representing.  Without both, you have compilation errors.

Or are you simply suggesting that the `ident` field be renamed to `name`?

Reply:
ah my thought is wrong, we indeed still need the ident. sorry.